Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -1328,3 +1340,486 @@ int CLuaEngineDefs::EngineResetSurfaceProperties(lua_State* luaVM)
+        case eObjectGroup::Modifiable::BUOYANCY:
+        case eObjectGroup::Modifiable::UPROOTLIMIT:
+        case eObjectGroup::Modifiable::COLDAMAGEMULTIPLIER:
+        case eObjectGroup::Modifiable::SMASHMULTIPLIER:
+        case eObjectGroup::Modifiable::BREAKVELOCITYRAND:
+        {
+            float fValue = g_GroupPropertiesGettersFloat[eProperty](pGroup);
+            lua_pushnumber(luaVM, fValue);
+            return 1;
+        }
+        case eObjectGroup::Modifiable::CAMERAAVOID:
+        case eObjectGroup::Modifiable::EXPLOSION:
+        case eObjectGroup::Modifiable::SPARKSONIMPACT:
+        {
+            bool bValue = g_GroupPropertiesGettersBool[eProperty](pGroup);
+            lua_pushboolean(luaVM, bValue);
+            return 1;
+        }
+        case eObjectGroup::Modifiable::FXOFFSET:
+        case eObjectGroup::Modifiable::BREAKVELOCITY:
+        {
+            CVector vecValue = g_GroupPropertiesGettersVector[eProperty](pGroup);
+            lua_pushvector(luaVM, vecValue);
+            return 1;
+        }
+        case eObjectGroup::Modifiable::COLDAMAGEEFFECT:
+        {
+            eObjectGroup::DamageEffect eDamEffect = pGroup->GetCollisionDamageEffect();
+            if (!EnumValueValid(eDamEffect))
+                break;
+
+            lua_pushstring(luaVM, EnumToString(eDamEffect));
+            return 1;
+        }
+        case eObjectGroup::Modifiable::SPECIALCOLRESPONSE:
+        {
+            eObjectGroup::CollisionResponse eColRepsonse = pGroup->GetCollisionSpecialResponseCase();
+            if (!EnumValueValid(eColRepsonse))
+                break;
+
+            lua_pushstring(luaVM, EnumToString(eColRepsonse));
+            return 1;
+        }
+        case eObjectGroup::Modifiable::FXTYPE:
+        {
+            eObjectGroup::FxType eFxType = pGroup->GetFxType();
+            if (!EnumValueValid(eFxType))
+                break;
+
+            lua_pushstring(luaVM, EnumToString(eFxType));
+            return 1;
+        }
+        case eObjectGroup::Modifiable::BREAKMODE:
+        {
+            eObjectGroup::BreakMode eBreakMode = pGroup->GetBreakMode();
+            if (!EnumValueValid(eBreakMode))
+                break;
+
+            lua_pushstring(luaVM, EnumToString(eBreakMode));
+            return 1;
+        }
+        case eObjectGroup::Modifiable::FXSYSTEM:
+        {
+            argStream.SetCustomError("Fx system name isn't possible to be extracted.");
+            break;
+        }
+    }
+
+    if (argStream.HasErrors())
+        m_pScriptDebugging->LogCustom(luaVM, argStream.GetFullErrorMessage());
+    lua_pushnil(luaVM);
+    return 1;
+}
+
+int CLuaEngineDefs::EngineRestoreObjectGroupPhysicalProperties(lua_State* luaVM)
+{
+    //  bool engineRestoreObjectGroupPhysicalProperties ( int groupID )
+    int iGivenGroup;
+
+    CScriptArgReader argStream(luaVM);
+    argStream.ReadNumber(iGivenGroup);
+
+    if (argStream.HasErrors())
```

Comment:
These errors should be changed accordingly (https://github.com/multitheftauto/mtasa-blue/issues/821#issuecomment-522187565).